### PR TITLE
Add build tool script and example tools with metadata and README files

### PR DESCRIPTION
**Pull request by guillevalin**
This pull request introduces several new scripts and updates to the Shinkai Tools repository to automate the building and running of tools. The changes include the addition of new bash scripts, workflow configurations, and updates to tool metadata and documentation.

#### New Scripts:
* [`.github/build_tool.sh`](diffhunk://#diff-0baa86d70143c74fae8c4d9ea61e944e989aa7e0933359d267d0c3a8d2225e73R1-R126): Added a script to automate the building of tools, including processing tool directories, creating jobs, and uploading tools to a node.
* [`.github/run_node.sh`](diffhunk://#diff-46baebc873e2f680ab200a4ef2ab0b85860f24b56adae9853ce5b5eaa5b42ec4R1-R38): Added a script to run the Shinkai node, including setting environment variables and downloading the appropriate node binary based on the platform.

#### Workflow Configuration:
* [`.github/workflows/build_tools.yaml`](diffhunk://#diff-3c29dd0ba4c3527a0384733964f46377bb9e36af318095b4cebdde8ce656f51eR1-R47): Added a GitHub Actions workflow to automate the building of Shinkai tools. This workflow includes steps to install dependencies, run the new scripts, and commit and push changes if any files are modified.

#### Tool Metadata and Documentation:
* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L1-R35): Updated the README to provide an overview of the available tools, their functionalities, and the project structure.
* `tools/hello-world-python/metadata.json`, `tools/hello-world-python/README.md`, `tools/hello-world-python/tool.py`: Added metadata, documentation, and implementation for the "Hello World Printer" tool. [[1]](diffhunk://#diff-3e41dfa552b1b3d0c17ee2c823dca1d83af5ba200aa68d3c97e0826dd53f2483R1-R32) [[2]](diffhunk://#diff-f8d5bfc2e37082daa018badddd9f05599b9d93fc8e76692f0c421829b8c55d71R1-R43) [[3]](diffhunk://#diff-983de395ae2950c6491456564415cf315b682b62c66bc6c05127488c819bad14R1-R21)
* `tools/hello-world/metadata.json`, `tools/hello-world/README.md`, `tools/hello-world/tool.ts`: Added metadata, documentation, and implementation for the "Hello World Function" tool. [[1]](diffhunk://#diff-5753a7355c7d8032f77cf5734ea6db9048b382e1245737a7fe606eca9a93755cR1-R32) [[2]](diffhunk://#diff-6b441e1f5a78fca3033830764fb742cbb734f7016adf253d3a090cb8a0dd6d6aR1-R38) [[3]](diffhunk://#diff-6fc146132ce08792aa06735d7e4c65838dc3833065dff3b2b2eb816a0df28fa7R1-R9)
* [`tools/smart-search/metadata.json`](diffhunk://#diff-c6eec6ec24b901bed426e15321a828e2fb386dae4f0583261e026d3849efdddbR2-R11): Updated metadata for the "Smart Search Engine" tool, including adding new keywords and tools. [[1]](diffhunk://#diff-c6eec6ec24b901bed426e15321a828e2fb386dae4f0583261e026d3849efdddbR2-R11) [[2]](diffhunk://#diff-c6eec6ec24b901bed426e15321a828e2fb386dae4f0583261e026d3849efdddbL24-L38) [[3]](diffhunk://#diff-c6eec6ec24b901bed426e15321a828e2fb386dae4f0583261e026d3849efdddbL98-R103)

These changes enhance the automation and documentation of the Shinkai Tools, making it easier to build, run, and understand the available tools.

### CODE REVIEW

File: build-tool.sh [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                },
+                false
+            ]
+            }' > "packages/${tool_name}/tool.json"
+
+        echo "Generated tool.json for ${tool_name}"
+
+        # Create zip file
+        cd "packages/${tool_name}"
+        zip "${toolkit_name}.zip" tool.json
```
Comment (acedward) [inline build-tool.sh:101]: creating the .zip here might be hard to maintain in the long term if we would have to keep in sync the node and this tool 👁️ the export endpoint can be called after v2/set_playground_tool

```
## Export Tool
curl "http://localhost:9950/v2/export_tool?tool_key_path=local%3A%3A%3Acheck_files2_shinkai%3A%3A%3Acheck_files2" \
     -H 'Authorization: Bearer debug' \
     -H 'Content-Type: application/json; charset=utf-8' > file.zip
```

File: tools/hello-world/metadata.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+    "author": "@@localhost.arb-sep-shinkai",
```
Comment (acedward) [inline tools/hello-world/metadata.json:2]:
```suggestion
    "author": "Shinkai",
```

File: build-tool.sh (hunk 1 above)
Comment (guillevalin) [reply build-tool.sh]: Created the .zip file directly from node with this approach. First I import it to the node and then I download it as a .zip file

**Review by acedward — APPROVED** [review]
🎖️

**Pull request merged**